### PR TITLE
Fix check for conf file ownership

### LINKED ISSUE #2550: Problems with 3.0.3-rc1
I noticed these problems with 3.0.3-rc1 so far:

1. The list of PRs doesn't exactly match the [3.0.3 RC milestone](https://github.com/sylabs/singularity/milestone/16): it left out pr #2535 and added pr #2533.

2. Pr #2533 breaks unprivileged mode and unprivileged installs.  It should only do the checks when not running in UserNamespace mode, if then.  Even running a root-installed singularity.conf with the exec -u option breaks.

Attn: @singularity-maintainers

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
**Description of the Pull Request (PR):**

Add fix on the check for ownership of config file, move check from
config.Parser() to actions.go after setuid check.
As stated by @DrDaveD , PR #2533 breaks unprivileged installations, and root-installed installation with the -u option.

Signed-off-by: Eduardo Arango <eduardo@sylabs.io>

**This fixes or addresses the following GitHub issues:**

- Fixes #2550 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: internal/app/singularity/actions.go [hunk 1]
```diff
@@ -553,6 +554,12 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 			sylog.Verbosef("starter-suid not found, using user namespace")
 			UserNamespace = true
 		}
+
+		// check for ownership of singularity.conf file when !UserNamespace
+		if !fs.IsOwner(configurationFile, 0) {
```
Comment (DrDaveD) [inline internal/app/singularity/actions.go]: Make this an 'else if' to work correctly when starter-suid is missing.  That's the case in an unprivileged installation that I maintain.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Review by DrDaveD — CHANGES_REQUESTED** [review]
If I do chown -R on /etc/singularity and run unprivileged, I get the error
```
FATAL:   /etc/singularity/capability.json must be owned by root
```

File: internal/app/singularity/actions.go (hunk 1 above)
Comment (cclerget) [reply internal/app/singularity/actions.go]: @ArangoGutierrez Those checks should definitively go in `PrepareConfig` function (`internal/pkg/runtime/engines/singularity/prepare.go`) to have something like this : 

```
	if starterConfig.GetIsSUID() {
		// check for ownership of singularity.conf
		if !fs.IsOwner(configurationFile, 0) {
			sylog.Fatalf("%s must be owned by root", configurationFile)
		}
	}
```

Same for `capability.json` but in `prepareUserCaps` function.

CLI won't prevent `starter-suid` from reading a configuration file not owned by root, that's why it must be done in `internal/pkg/runtime/engines/singularity/prepare.go`

**Review by DrDaveD — APPROVED** [review]
Looks good to me now.  I tried on el 7.6, with the commits from this PR cherry-picked into a branch of release-3.0.  I tried a chown -R on /etc/singularity with a regular root install and exec -u, and tried an unprivileged install in a scratch directory with libexec/singularity/bin/starter-suid removed.

**Review by cclerget — APPROVED** [review]
LGTM

**Review by jmstover — APPROVED** [review]

**Pull request merged**